RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc

```diff
@@ -420,10 +421,14 @@ bool RecHitTools::isSilicon(const DetId& id) const {
   bool issilicon = false;
   if (id.det() == DetId::HGCalEE || id.det() == DetId::HGCalHSi)
     issilicon = true;
+  if (id.det() == DetId::Forward && id.subdetId() == static_cast<int>(HFNose)) {
```

Comment:
the entire isSilicon function can be simplified to:
```
return (id.det() == DetId::HGCalEE or id.det() == DetId::HGCalHSi or (id.det() == DetId::Forward && id.subdetId() == static_cast<int>(HFNose)));
```

Reply:
done